content/blog/eng/lambda-days-2020.md

```diff
@@ -0,0 +1,91 @@
+---
+authors: ["andreac"]
+comments: true
+date: "2020-04-09"
+draft: false
+share: true
+categories: [English, Conference, Lambda-days, Krakow, Functional programming]
+title: "Lambda-days-2020"
+ita: "Lambda-days-2020"
+
+languageCode: "en-US"
+type: "post"
+---
+
+# Lambda Days 2021
+
+This year, Facile attended for the first time to [Lambda Days](https://www.lambdadays.org/lambdadays2020) Conference in the beautiful polish city of Krakow.
+It has been a two day full immersion, where the organizers have been able to gather speakers from all over the world
+showcasing the latest trends and academic research in the functional world, spanning from beginner approaches to functional languages 
+to super specific scientific applications.
+
+This conference totally worth attending, albeit very intense (sometimens mind blowing, actually!), the talks were high quality, and perfect organization.
+
+Here is a short list of talks that took our attention:
+
+- **Keynote: How to specify it! A guide to writing properties of pure functions**
+
+[Video](https://www.youtube.com/watch?v=G0NUOst-53U&list=PLvL2NEhYV4ZsV9Bw0wp1P46SOdtk4pFW6&index=3&t=0s)
+
+**John Hughes** one of Haskell creators presented his QuickCheck library
+In a funny manner, he showed how use his library to create automated test-cases and gave some tips on how to write tests that actually
+verify your assumptions and not giving false green check.
+He suggested first to write pure function as much as you can and creating properties by which test conditions can be automated.
+
+
+- **The power of Π**
+
+**Thorsten Altenkirch** from the University of Nottingham gave a talk about implementing dependent types
+Think for a second if it was possible to base return types in functions by the value at runtime. Normally you would think is not possible
+because we should be at runtime to know the actual input vale. The truth is that with some mathematical assumptions is possible to obtain
```

Comment:
The truth is that with some mathematical assumptions, it is possible to obtain types such as non-empty arrays that otherwise a compiler would not know.

**Note:** I'm not sure about the verb to know at the end of the sentence. What is the exact meaning of the sentence in italian?

Reply:
Poniamo che una *funzione* accetti in input una lista di valori, ad esempio  un array. L'array, a priori, non porta con sè l'informazione del numero di elementi che contiene, lo si conosce solo se si esegue il codice.
L'idea quindi è che facendo alcune assunzioni, andiamo a conoscere queste informazioni prima dell'esecuzione del codice ed eliminare una serie di problemi come il dover considerare una lista vuota ecc
Questo è un articolo un po' lungo ma abbastanza chiaro, se ti interessa:
https://medium.com/background-thread/the-future-of-programming-is-dependent-types-programming-word-of-the-day-fcd5f2634878

Reply:
@Glitches thank you for the suggestion! 
I think the verb "to know" is right, then. 
I would just rephrase the sencence on line 40 a little bit (see comment above)